### PR TITLE
[FIX]Print fixes for rc/2022-july

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fixed prints when fetching old/unnecessary hb v2 messages + cache cleanup
- better handling when logging errors in historyRepository, removed some duplicated code

### CODE REVIEW

File: errors/closingError.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+)
+
+// IsClosingError returns true if the provided error is used whenever the node is in the closing process
+func IsClosingError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	return errors.Is(err, ErrContextClosing) ||
+		errors.Is(err, ErrDBIsClosed) ||
```
Comment (sasurobert) [inline errors/closingError.go:15]: no need for the first 2 conditions as 3rd and 4th will resolve the issues.

**Review by sstanculeanu — COMMENTED** [review]
looks good, only Roberts' note

**Review by BeniaminDrasovean — APPROVED** [review]

File: state/storagePruningManager/storagePruningManager.go [hunk 2]
```diff
@@ -170,7 +169,7 @@ func (spm *storagePruningManager) prune(rootHash []byte, tsm common.StorageManag
 
 	err := spm.removeFromDb(rootHash, tsm, handler)
 	if err != nil {
-		if err == storage.ErrDBIsClosed || strings.Contains(err.Error(), storage.ErrDBIsClosed.Error()) {
+		if errors.IsClosingError(err) {
```
Comment (andreibancioiu) [inline state/storagePruningManager/storagePruningManager.go:172]: :rocket:

**Review by andreibancioiu — APPROVED** [review]

File: errors/closingError.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+		strings.Contains(err.Error(), ErrDBIsClosed.Error()) ||
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply errors/closingError.go:16]: simplified as suggested

File: dblookupext/errors.go [hunk 4]
```diff
@@ -31,3 +34,11 @@ func newErrCannotSaveEpochByHash(what string, hash []byte, originalErr error) er
 func newErrCannotSaveMiniblockMetadata(hash []byte, originalErr error) error {
 	return fmt.Errorf("cannot save miniblock metadata, hash [%s]: %w", hex.EncodeToString(hash), originalErr)
 }
+
+func handleErrorLogging(logInstance logger.Logger, logLevel logger.LogLevel, err error, message string, args ...interface{}) {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply dblookupext/errors.go:38]: agree, we can add custom handlers for a function in the logger like:
```
LogError(logger.Level, message, error, extraArgs...)
```

**Review by andreibancioiu — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

File: errors/closingError.go (hunk 1 above)
Comment (iulianpascalau) [reply errors/closingError.go]: done

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by popenta — APPROVED** [review]
@@                            Log scanner                                  @@
##                            print-fixes                                  ##
================================================================================
- Known Warnings              20
- New Warnings                0
- Known Errors                0
- New Errors                  1
- Panics                      0
================================================================================

**Pull request merged**